project/Boilerplate.scala

```diff
@@ -500,4 +501,34 @@ object Boilerplate {
     }
   }
 
+  object GenTupleShowInstances extends Template {
+    def filename(root: sbt.File): File =
+      root / "cats" / "instances" / "NTupleInstances.scala"
+
+    def content(tv: TemplateVals): String = {
+      import tv._
+
+      def constraints(name: String) =
+        synTypes.map(tpe => s"$tpe: $name[$tpe]").mkString(", ")
+
+      val showMethod: String =
+        synTypes.zipWithIndex.iterator
+          .map {
+            case (tpe, i) => s"$${${tpe}.show(f._${i + 1})}"
+          }
+          .mkString("s\"(", ",", ")\"")
+
+      block"""
+      |
+      |package cats
+      |package instances
+      |
+      |private[instances] trait NTupleInstances {
+      -  implicit final def catsShowForTuple$arity[${`A..N`}](implicit ${constraints("Show")}): Show[${`(A..N)`}] =
```

Comment:
Could we call these `catsStdShowForTupleN` to match the standard instance naming, even though that means we'd either have to make `Tuple2Instances` extend `cats.instances.NTupleInstances` or special case `N = 2` here (the better choice in my view)?

Reply:
Sure I will do it. That (the `N=2` case) is exactly the reason why I changed from the standard 😄 . I will add the exception case for 2 and a comment.